tensordict/functional.py

```diff
@@ -87,76 +87,92 @@ def pad_sequence(
 
     Args:
         list_of_tensordicts (List[TensorDictBase]): the list of instances to pad and stack.
-        batch_first (bool, optional): the ``batch_first`` correspondant of :func:`torch.nn.utils.rnn.pad_sequence`.
-            Defaults to ``True``.
+        pad_dim (int, optional): the ``pad_dim`` indicates the dimension to pad all the keys in the tensordict.
```

Comment:
Shouldn't we default to -1? Or tensordict.ndim (to pad the first dim after batch size)?
Also I would make it clear that we're talking about tensor dimensions, not tensordict's

Reply:
I think the most common application for padding tensors is in sequential data. For example, the transformer expects the input data to be in the form:
`src: (S,E) for unbatched input, (N, S, E) or (S, N, E)`
where S is the sequence dimension, E is the embedding dimension, and N is the batch dim. Since in our case we stack the tensordicts into a new dimension, I think our tensordicts are supposed to be a collection of the form (S, E). Thus, I think it would be a good option to default to `pad_dim=0`.

Reply:
My issue with that is that tensordict is designed to work with data that comes in batches of arbitrary dimensions. Defaulting to 0 will cover only a small subset of cases (to me we should raise an exception any time the dimension is within the number of batch dimensions).
We could say None by default which means defaulting to input.ndim wdyt?

Reply:
Since we are stacking the TensorDicts all the tensors for each key must have the same dimensionality except the `pad_dim`. By stacking we create a new dimension which I interpret as the batch_size. We can consider to generate a more advanced operator which is somewhat equivalent to `torch.cat` but also pads every other dimension so we can handle more arbitrary sizes.